Source/DafnyLanguageServer/Language/IDiagnosticPublisher.cs

```diff
@@ -10,7 +10,7 @@ public interface IDiagnosticPublisher {
     /// Publishes the diagnostics of the specified dafny document to the connected LSP client.
     /// </summary>
     /// <param name="document">The document whose diagnostics should be published.</param>
-    void PublishDiagnostics(DafnyDocument document, CancellationToken cancellationToken);
```

Comment:
Are we free to break public APIs in this project?

Reply:
The C# API is not a dependency of other projects, right? It does provide the LSP API, and some custom notifications layered over that, to the VSCode Dafny extension.

Reply:
It was an honest question. :) I'm not aware of any other consumers.